code/src/containers/forms/new-product-form/NewProductForm.test.tsx

```diff
@@ -0,0 +1,328 @@
+  useCreateProductMutation: () => [
+    mockCreateProductMutation,
+    { isLoading: false }
+  ]
+}));
+
+jest.mock("@/hooks/use-snackbar/useSnackbar", () => ({
+  __esModule: true,
+  default: () => ({ openSnackbarWithTimeout: mockShowSnackbar })
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockNavigate
+}));
+
+const testData = {
+  status: "HIDDEN",
+  image: "https://example.com",
+  quantity: 10,
+  price: 100,
+  tagIds: [3],
+  productTranslations: [
+    {
+      name: "name",
+      description: "description-description",
+      languageCode: "en"
+    }
+  ]
+};
+
+const twoTranslationsTestData = {
+  ...testData,
+  productTranslations: [
+    testData.productTranslations[0],
+    { ...testData.productTranslations[0], languageCode: "uk" }
+  ]
+};
+
+describe("Test NewProductForm", () => {
+  let imgUrlInput: HTMLInputElement;
+  let priceInput: HTMLInputElement;
+  let quantityInput: HTMLInputElement;
+  let categorySelect: HTMLSpanElement;
+  let nameInput: HTMLInputElement;
+  let descriptionInput: HTMLTextAreaElement;
+  let submitButton: HTMLButtonElement;
+  let statusInput: HTMLInputElement;
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const render = () => {
+    const result = renderWithProviders(<NewProductForm />);
+
+    imgUrlInput = screen
+      .getByTestId("new-product-image-input")
+      .querySelector("input")!;
+    priceInput = screen
+      .getByTestId("new-product-price-input")
+      .querySelector("input")!;
+    quantityInput = screen
+      .getByTestId("new-product-quantity-input")
+      .querySelector("input")!;
+    categorySelect = screen.getByLabelText("productForm.inputLabel.category");
+    nameInput = screen
+      .getByTestId(`new-product-name-input-en`)
+      .querySelector("input")!;
+    descriptionInput = screen
+      .getByTestId(`new-product-description-input-en`)
+      .querySelector("textarea")!;
+    statusInput = screen
+      .getByTestId("new-product-status-checkbox")
+      .querySelector("input")!;
+    submitButton = screen.getByText("productForm.submit");
+
+    return result;
+  };
+
+  const fillInTestData = async () => {
+    await typeIntoInput(imgUrlInput, testData.image);
+    await typeIntoInput(priceInput, testData.price);
+    await typeIntoInput(quantityInput, testData.quantity);
+    await typeIntoInput(nameInput, testData.productTranslations[0].name);
+    await typeIntoInput(
+      descriptionInput,
+      testData.productTranslations[0].description
+    );
+  };
+
+  const submit = async () => {
+    await act(async () => {
+      fireEvent.click(submitButton);
+    });
+  };
+
+  const selectCategory = async () => {
+    await act(async () => {
+      fireEvent.mouseDown(categorySelect);
+    });
+
+    const categoryOption = screen.getByText("productsAll.computer");
+
+    await act(async () => {
+      fireEvent.click(categoryOption);
+    });
+  };
```

Comment:
Can we move it outside `describe` block?

Reply:
Yes, we can. I moved it outside